pkg/sentry/kernel/task_signals.go

```diff
@@ -263,6 +263,18 @@ func (t *Task) deliverSignalToHandler(info *arch.SignalInfo, act arch.SignalAct)
 	if t.haveSavedSignalMask {
 		mask = t.savedSignalMask
 	}
+
+	// Restorer is not required on arm64,
+	// we borrow it to store the value of AT_SYSINFO_EHDR.
+	if arch.Host.String() == "arm64" {
+		for i := (len(t.MemoryManager().Auxv()) - 1); i >= 0; i-- {
+			key := uint64(t.MemoryManager().Auxv()[i].Key)
+			if key == linux.AT_SYSINFO_EHDR {
```

Comment:
If we need the address of the VDSO, I think it would be cleaner to just store it directly in the MemoryManager, rather than searching auxv.

Reply:
Done. I have finished it in https://github.com/google/gvisor/pull/1978